bindings/cpp/session.cpp

```diff
@@ -2233,12 +2248,14 @@ error_info session::mix_states(const key &id, std::vector<int> &groups)
 }
 
 async_iterator_result session::start_iterator(const key &id, const std::vector<dnet_iterator_range>& ranges,
-								uint32_t type, uint64_t flags,
-								const dnet_time& time_begin, const dnet_time& time_end)
+		uint32_t type, uint64_t flags, const dnet_time& time_begin, const dnet_time& time_end)
 {
-	auto ranges_size = ranges.size() * sizeof(ranges.front());
+	size_t ranges_size = 0;
+	if (ranges.size() != 0)
+		ranges_size = ranges.size() * sizeof(ranges.front());
```

Comment:
This is not about sizeof(), but about ranges.front(), which may fail.
Consider this as following example
```
int a[3] = {0, 1, 2};
printf("%d\n", sizeof(a[12]))'
```

Will it compile? Sure.
Will it fail? Probably not, since a[12] may or may not be dereferenced.

To make things clear I made those changes.